Tools/TestWebKitAPI/Tests/WebKitCocoa/WebsitePolicies.mm

```diff
@@ -1833,3 +1835,148 @@ - (void)webView:(WKWebView *)webView runJavaScriptAlertPanelWithMessage:(NSStrin
     [preferences _setNetworkConnectionIntegrityEnabled:YES];
     EXPECT_TRUE([preferences _networkConnectionIntegrityEnabled]);
 }
+
+static BOOL overrideHandlesURLScheme()
+{
+    return NO;
+}
+
+TEST(WebpagePreferences, ExtensionPageContentBlockers)
+{
+    ClassMethodSwizzler swizzler([WKWebView class], @selector(handlesURLScheme:), reinterpret_cast<IMP>(overrideHandlesURLScheme));
+
+    [[WKContentRuleListStore defaultStore] _removeAllContentRuleLists];
+
+    auto navigationDelegate = adoptNS([TestNavigationDelegate new]);
+    navigationDelegate.get().decidePolicyForNavigationActionWithPreferences = ^(WKNavigationAction *action, WKWebpagePreferences *preferences, void (^decisionHandler)(WKNavigationActionPolicy, WKWebpagePreferences *)) {
+        [preferences _setContentBlockersEnabled:YES];
+        if (action.targetFrame.mainFrame)
+            [preferences _setContentBlockersEnabled:NO];
+
+        decisionHandler(WKNavigationActionPolicyAllow, preferences);
+    };
+
+    auto handler = adoptNS([TestURLSchemeHandler new]);
+    [handler setStartURLSchemeTaskHandler:^(WKWebView *, id<WKURLSchemeTask> task) {
+        NSString *path = task.request.URL.path;
+        NSString *type = nil;
+        NSString *result = nil;
+        if ([path hasSuffix:@"index.html"]) {
+            result = @"<script>window.results = [];</script>"
+            "<script src='test:///script.js'></script>"
+            "<iframe src='test:///subframe.html'></iframe>";
+            type = @"text/html";
+        } else if ([path hasSuffix:@"subframe.html"]) {
+            result = @"<script src='test:///script_subframe.js'></script>";
+            type = @"text/html";
+        } else if ([path hasSuffix:@"script.js"]) {
+            result = @"window.results.push(window.location.href);";
+            type = @"text/javascript";
+        } else if ([path hasSuffix:@"script_subframe.js"]) {
+            result = @"window.parent.results.push(window.location.href);";
+            type = @"text/javascript";
+        }
+
+        if (!result) {
+            [task didFailWithError:[NSError errorWithDomain:@"TestWebKitAPI" code:1 userInfo:nil]];
+            return;
+        }
+
+        auto response = adoptNS([[NSURLResponse alloc] initWithURL:task.request.URL MIMEType:type expectedContentLength:[result length] textEncodingName:nil]);
+        [task didReceiveResponse:response.get()];
+        [task didReceiveData:[result dataUsingEncoding:NSUTF8StringEncoding]];
+        [task didFinish];
+    }];
+
+    auto configuration = adoptNS([[WKWebViewConfiguration alloc] init]);
+    [configuration setURLSchemeHandler:handler.get() forURLScheme:@"safari-extension"];
```

Comment:
Let's not use this scheme in the tests.  WebKit should know nothing about this scheme.

Reply:
Changed the scheme to `test`.